### PR TITLE
Make JWT command uses config connection details

### LINKED ISSUE #1744: SNOW-1752069: generate-jwt should use settings from config.toml
### Description

Currently `generate_jwt` command has its own set of flags and does not seem to take `config.toml` into account at all.

For example:

```
snow connection generate-jwt --private-key <p8-file-path>                             
Enter private key file password (Press enter if none) []: 

An unexpected exception occurred. Use --debug option to see the traceback. Exception message:

argument of type 'NoneType' is not iterable
```
The error
>argument of type 'NoneType' is not iterable
Probably happens because of the type [here](https://github.com/snowflakedb/snowflake-cli/blob/main/src/snowflake/cli/_plugins/connection/commands.py#L360):

```python
def generate_jwt(
    account: str = typer.Option(
        None,
        "--account",
        "-a",
        "--accountname",
        help="Account name to use when authenticating with Snowflake.",
        show_default=False,
    ),
```

This also applies to the `user` and `private_key_file` values.

* Other commands use values from the configuration file
* `argument of type 'NoneType' is not iterable` is not immediately apparent on how to fix. `--debug` does show the location, but the error message could be better

### Context

_No response_

### PR DESCRIPTION
**Pull request by sfc-gh-turbaszek**
### Pre-review checklist
   * [x] I've confirmed that instructions included in README.md are still correct after my changes in the codebase.
   * [x] I've added or updated automated unit tests to verify correctness of my new code.
   * [x] I've added or updated integration tests to verify correctness of my new code.
   * [x] I've confirmed that my changes are working by executing CLI's commands manually on MacOS.
   * [ ] I've confirmed that my changes are working by executing CLI's commands manually on Windows.
   * [x] I've confirmed that my changes are up-to-date with the target branch.
   * [ ] I've described my changes in the release notes.
   * [x] I've described my changes in the section below.

### Changes description
Closes #1744

### CODE REVIEW

File: src/snowflake/cli/_plugins/connection/commands.py [hunk 1]
```diff
@@ -355,36 +354,23 @@ def set_default(
...
+
+    msq_template = (
+        "{} is not set in the connection context, but required for JWT generation."
+    )
+    if not connection_details.user:
+        raise UsageError(msq_template.format("User"))
+    if not connection_details.account:
+        raise UsageError(msq_template.format("Account"))
+    if not connection_details.private_key_file:
+        raise UsageError(msq_template.format("Private key file"))
```
Comment (sfc-gh-astus) [inline src/snowflake/cli/_plugins/connection/commands.py:372]: No password?

Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/_plugins/connection/commands.py:372]: What do you mean?

Comment (sfc-gh-astus) [reply src/snowflake/cli/_plugins/connection/commands.py:372]: We are not able to provide password? We have to use private key?

**Review by sfc-gh-astus — APPROVED** [review]

File: src/snowflake/cli/_plugins/connection/commands.py (hunk 1 above)
Comment (sfc-gh-turbaszek) [reply src/snowflake/cli/_plugins/connection/commands.py:372]: Yes, only private key is neccessary

**Pull request merged**